war/src/main/java/io/lumeer/engine/controller/SecurityFacade.java

```diff
@@ -51,9 +52,377 @@
    private DataStorage dataStorage;
 
    @Inject
+   @SystemDataStorage
+   private DataStorage systemDataStorage;
+
+   @Inject
    private DataStorageDialect dataStorageDialect;
 
    @Inject
+   private UserFacade userFacade;
+
+   /********* CHECKING ROLES *********/
+
+   public boolean hasOrganizationRole(String organizationId, String role) {
+      DataDocument doc = systemDataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ORGANIZATION_ROLES_COLLECTION_NAME,
+            dataStorageDialect.fieldValueFilter(LumeerConst.Security.ORGANIZATION_ID_KEY, organizationId),
+            Arrays.asList(LumeerConst.Security.ROLES_KEY));
+
+      return checkRole(doc, userFacade.getUserEmail(), role);
+   }
+
+   public boolean hasProjectRole(String projectId, String role) {
+      DataDocument doc = dataStorage.readDocumentIncludeAttrs(
+            LumeerConst.Security.ROLES_COLLECTION_NAME,
+            dataStorageDialect.multipleFieldsValueFilter(projectFilter(projectId)),
+            Arrays.asList(dataStorageDialect.concatFields(LumeerConst.Security.ROLES_KEY, role)));
+
+      return checkRole(doc, userFacade.getUserEmail(), role);
+   }
+
+   public boolean hasCollectionRole(String collectionName, String projectId, String role) {
```

Comment:
Just a detail but I would rather but `projectId` parameter first since it is higher in the hierarchy.